rewrite-java/src/main/java/org/openrewrite/java/InlineMethodCalls.java

```diff
@@ -0,0 +1,309 @@
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
+                InlineMeValues values = findInlineMeValues(mi.getMethodType());
+                if (values == null) {
+                    return mi;
+                }
+                Template template = values.template(mi);
+                if (template == null) {
+                    return mi;
+                }
+                removeAndAddImports(method, values.getImports(), values.getStaticImports());
+                J replacement = JavaTemplate.builder(template.getString())
+                        .contextSensitive()
+                        .imports(values.getImports().toArray(new String[0]))
+                        .staticImports(values.getStaticImports().toArray(new String[0]))
+                        .javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))
+                        .build()
+                        .apply(updateCursor(mi), mi.getCoordinates().replace(), template.getParameters());
+                return avoidMethodSelfReferences(mi, replacement);
+            }
+
+            @Override
+            public J visitNewClass(J.NewClass newClass, ExecutionContext ctx) {
+                J.NewClass nc = (J.NewClass) super.visitNewClass(newClass, ctx);
+                InlineMeValues values = findInlineMeValues(nc.getConstructorType());
+                if (values == null) {
+                    return nc;
+                }
+                Template template = values.template(nc);
+                if (template == null) {
+                    return nc;
+                }
+                removeAndAddImports(newClass, values.getImports(), values.getStaticImports());
+                J replacement = JavaTemplate.builder(template.getString())
+                        .contextSensitive()
+                        .imports(values.getImports().toArray(new String[0]))
+                        .staticImports(values.getStaticImports().toArray(new String[0]))
+                        .javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))
+                        .build()
+                        .apply(updateCursor(nc), nc.getCoordinates().replace(), template.getParameters());
+                return avoidMethodSelfReferences(nc, replacement);
+            }
+
+            private @Nullable InlineMeValues findInlineMeValues(JavaType.@Nullable Method methodType) {
+                if (methodType == null) {
+                    return null;
+                }
+                List<String> parameterNames = methodType.getParameterNames();
+                if (!parameterNames.isEmpty() && "arg0".equals(parameterNames.get(0))) {
+                    return null; // We need `-parameters` before we're able to substitute parameters in the template
```

Comment:
This feels like something we could always enable in `JavaParser` if it isn't by default

Reply:
Yes it's here mostly as a safeguard; it should be set in each of the Java parsers we have:
https://github.com/openrewrite/rewrite/blob/380c94278772192b2c71a02f41dfeee057ec5e29/rewrite-java-21/src/main/java/org/openrewrite/java/isolated/ReloadableJava21Parser.java#L112